env0/resource_environment_test.go

```diff
@@ -81,6 +88,118 @@ func TestUnitEnvironmentResource(t *testing.T) {
 		})
 	})
 
+	// TODO: test deploy with variables ( needs configuration changes read )
+
+	t.Run("Update to: template id, revision, configuration should trigger a deployment", func(t *testing.T) {
+		environment := client.Environment{
+			Id:        "id0",
+			Name:      "my-environment",
+			ProjectId: "project-id",
+			LatestDeploymentLog: client.DeploymentLog{
+				BlueprintId:       "template-id",
+				BlueprintRevision: "revision",
+			},
+		}
+		updatedEnvironment := client.Environment{
+			Id:        updatedEnvironment.Id,
+			Name:      environment.Name,
+			ProjectId: environment.ProjectId,
+			LatestDeploymentLog: client.DeploymentLog{
+				BlueprintId:       "updated template id",
+				BlueprintRevision: "updated revision",
+			},
+		}
+
+		varType := client.ConfigurationVariableTypeEnvironment
+		varSchema := client.ConfigurationVariableSchema{
+			Type: "string",
+			Enum: []string{"a", "b"},
+		}
+		configurationVariables := client.ConfigurationVariable{
+			Value:  "my env var value",
+			Name:   "my env var",
+			Type:   &varType,
+			Schema: &varSchema,
+		}
+
+		updatedEnvironmentResource := fmt.Sprintf(`
+				resource "%s" "%s" {
+					name = "%s"
+					project_id = "%s"
+					template_id = "%s"
+					revision = "%s"
+					configuration {
+						name = "%s"
+						value = "%s"
+						schema_type = "%s"
+						schema_enum = ["%s"]
+					}
+				}`,
+			resourceType, resourceName, environment.Name,
+			updatedEnvironment.ProjectId, updatedEnvironment.LatestDeploymentLog.BlueprintId,
+			updatedEnvironment.LatestDeploymentLog.BlueprintRevision, configurationVariables.Name,
+			configurationVariables.Value, configurationVariables.Schema.Type,
+			strings.Join(configurationVariables.Schema.Enum, "\",\""),
+		)
```

Comment:
I used sprintF instead of the resource build helper because
I started writing this test with config variables in mind
The builder doesn't support resources inside resources
I left it like this so its easier to add them in the. upcoming pr

resource with configuration changes example: 
```
resource "env0_environment" "test" {
  name        = "env name"
  project_id  = "ed7c54d2-ce49-4ad9-ab4e-be5f62107f03"
  template_id = "6a030803-4643-4ad0-a1fe-037feed14f21"
  configuration {
  	name = "var_name"
  	value = "a"
    schema_type = "string"
    schema_enum = ["a","b","c"]
  }
}
```

Reply:
ended up testing the vars so Im resolving this